packages/api/src/controllers/stream.ts

```diff
@@ -193,13 +195,42 @@ export function getPlaybackUrl(ingest: string, stream: DBStream) {
   return pathJoin(ingest, `hls`, stream.playbackId, `index.m3u8`);
 }
 
-function getRecordingUrl(ingest: string, session: DBSession, mp4 = false) {
-  return pathJoin(
+async function getRecordingUrls(
+  config: CliArgs,
+  ingest: string,
+  session: DBSession
+): Promise<{ recordingUrl: string; mp4Url: string }> {
+  const asset = await db.asset.getBySessionId(session.id);
+  if (!asset || ["waiting", "processing"].includes(asset.status?.phase)) {
+    // Recording processing in progress
+    return;
+  }
+
+  if (asset.id != session.id) {
```

Comment:
Neat :)

**nit:**
```suggestion
  if (asset.id !== session.id) {
```

Also, another way to check if it's an old pipeline asset is through the `asset.playbackRecordingId` field. I think it might be more specific so slightly preferrable. Really soft opinion tho, it's fine like this.

Reply:
I think, we can't use `playbackRecordingId` here, because it's set in task-runner only if the recording succeeded. For failed assets we don't have `playbackRecordingId`.